File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int[] nums = {0, 0};
        int idx = 0;

        for (int student: students) {
            nums[student]++;
        }

        while (idx < sandwiches.length) {
            if (nums[sandwiches[idx]] > 0) {
                nums[sandwiches[idx]]--;
            } else {
                break;
            }
            idx++;
        }

        return sandwiches.length - idx;
    }
}

File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        nums = [0, 0]
        idx = 0

        for student in students:
            nums[student] += 1

        while idx < len(sandwiches):
            if nums[sandwiches[idx]] > 0:
                nums[sandwiches[idx]] -= 1
            else:
                break
            idx += 1

        return len(sandwiches) - idx